solidity/contracts/AugurConstantProductMarket.sol

```diff
@@ -148,6 +146,24 @@ contract AugurConstantProduct is ERC20 {
 		shareToken.unsafeBatchTransferFrom(from, to, tokenIds, amounts);
 	}
 
+	// formula where x = yes shares owned when swapping yes
+	// s = (nf + 1000x -xf - 1000y - 1000n + sqrt(x^2f^2 - 2000x^2f + 1000000x^2 - 2000000nx - 2nxf^2 + 4000nxf + 2000000xy - 2000xfy + 1000000n^2 + n^2f^2 - 2000n^2f + 1000000y^2 + 2000000ny - 2000nfy)) / 2(1000 - f)
+	// s = swap amount
+	// x = yes shares owned
+	// n = pool no
+	// y = pool yes
+	// f = fee
+	function getCloseOutSwapAmount(uint256 shareBalance, bool swapYes) external returns (uint256) {
+		int256 n = swapYes ? int256(noBalance) : int256(yesBalance);
+		int256 y = swapYes ? int256(yesBalance) : int256(noBalance);
+		int256 x = int256(shareBalance);
+		int256 f = int256(fee);
```

Comment:
Everything is positive to start with, and we get a positive number out.  Why are we doing all of the math with signed integers?  Is the problem that some step along the way can go negative?